test/worker_test.py

```diff
@@ -1042,6 +1042,62 @@ def test_stop_getting_new_work_assistant_keep_alive(self):
         self._test_stop_getting_new_work(keep_alive=True, assistant=True)
 
 
+class WorkerLockTest(LuigiTestCase):
+
+    class LockTask(RunOnceTask):
+
+        name = luigi.Parameter()
+        delay1 = luigi.FloatParameter()
+        delay2 = luigi.FloatParameter()
+        use_locking = luigi.BoolParameter()
+        log_file = luigi.Parameter()
+
+        def write_line(self, line):
+            with open(self.log_file, "a") as f:
+                f.write(line + "\n")
+
+        def run(self):
+            time.sleep(self.delay1)
+            if self.use_locking:
+                self.worker_lock.acquire()
```

Comment:
So if you don't acquire and release the lock, there are no functional changes? (i.e. none impacting to existing users who don't explicitly use this feature)

Reply:
Exactly.